### ISSUE TITLE
On hyperparameters for Table 3.

### ISSUE THREAD

**Issue by honggudrnjs** [ZexiLee/ICML-2023-FedLAW#4]
Thanks for sharing the code of this valuable work.
Following the results of your work, I wonder hyperparameters for reproduce of Table 3.
Currently, the hyperparameter setting in 'args.py' does not fit to Table , because E is set to 1 and server_valid_ratio is different from what is explained in the paper.
I followed the details in the paper like the following:
     CIFAR-10, fmnist vs CIFAR-100
LR             0.08         vs        0.01
wd            5e-4         vs       5e-5
server_valid_ratio    0.01   vs    0.1 (10*(# of classes) / (# of test data))

However, we found that running FedLAW with CNN (w/ Dirichlet alpha 0.1, CIFAR-100) produces random prediction.
Thus, we wonder if we missed some details. Is client_valid_ratio 0.3 right? or server_valid_ratio or server_epochs should be changed?

**Comment by ZexiLee** [comment]
Thanks for this raised issue.

I guess "produces random prediction" means that FedLAW works well in the early but suddenly crashes into the random prediction model. 
I met these situations before and not only happened to FedLAW but also FedAvg and other methods. I think this is due to severe model drifts. 
I suggest you slightly tune server_valid_ratio, server_epochs, server_lr, or other hyperparameters for FedLAW. Sometimes, changing another random seed is just fine. Also, reducing the local epochs and learning rates may also help.

Hoping my reply can help you.

**Comment by honggudrnjs** [comment]
Thanks for your reply. Could you check dirichlet alpha one more time? I guess dirichlet alpha for non-IID is actually 1.0, not 0.1 as in Table 3. In CNN and CIFAR-100 setting, FedLAW is not working for alpha=0.1, but is working for alpha=1.0 with similar performance in Table 3.
